--- a/src/Makefile	2019-08-25 15:02:16.000000000 +0100
+++ b/src/Makefile	2019-09-09 23:33:39.628526053 +0100
@@ -26,6 +26,7 @@
 
 include $(UTILS_MK_DIR)/utils.mk
 
+XNVCTRL_MAKEFILE ?= Makefile
 XNVCTRL_DIR ?= ../src/libXNVCtrl
 
 include $(XNVCTRL_DIR)/xnvctrl.mk
@@ -301,2 +302,8 @@
 
+# define the rule to build $(XNVCTRL_ARCHIVE)
+$(XNVCTRL_ARCHIVE): build-xnvctrl
+
+build-xnvctrl:
+	@$(MAKE) -C $(XNVCTRL_DIR) -f $(XNVCTRL_MAKEFILE)
+
 clean clobber:
