--- a/main.cpp	2016-08-29 22:26:53.280246719 +0100
+++ b/main.cpp	2016-08-30 00:22:24.079748597 +0100
@@ -49,1 +49,2 @@
 #include <unistd.h>
+#include <sys/types.h>
@@ -57,2 +58,10 @@
     {
+	int user_id=getuid();
+	if ( user_id == 0 ) {
+#ifdef DEBUG_KDE
+        std::cerr << "User ID=" << user_id << std::endl;
+#endif
+		return 1;
+	}
+
     KAboutData about( "kmozillahelper", "kdelibs4", ki18n( "KMozillaHelper" ), APP_HELPER_VERSION );
