--- a/main.cpp	2017-02-11 01:09:53.657026496 +0000
+++ b/main.cpp	2017-02-11 01:26:01.124398705 +0000
@@ -26,6 +26,7 @@
 
 #include <cassert>
 #include <unistd.h>
+#include <sys/types.h>
 
 #include <iostream>
 
@@ -57,6 +58,14 @@
 
 int main(int argc, char* argv[])
 {
+
+    int user_id=getuid();
+    if ( user_id == 0 ) {
+#ifdef DEBUG_KDE
+        std::cerr << "User ID=" << user_id << std::endl;
+#endif
+        return 1;
+    }
     QApplication app(argc, argv);
 
     // This shows on file dialogs
