--- a/cmake/CUDA.cmake	2022-04-04 12:59:17.000000000 +0100
+++ b/cmake/CUDA.cmake	2023-10-19 15:26:33.194129516 +0100
@@ -16,18 +16,6 @@
 
 set(DEFAULT_CUDA_ARCH "50")
 
-# Fermi GPUs are only supported with CUDA < 9.0
-if (CUDA_VERSION VERSION_LESS 9.0)
-    list(APPEND DEFAULT_CUDA_ARCH "20;21")
-endif()
-
-# Kepler GPUs are only supported with CUDA < 11.0
-if (CUDA_VERSION VERSION_LESS 11.0)
-    list(APPEND DEFAULT_CUDA_ARCH "30")
-else()
-    list(APPEND DEFAULT_CUDA_ARCH "35")
-endif()
-
 # add Pascal support for CUDA >= 8.0
 if (NOT CUDA_VERSION VERSION_LESS 8.0)
     list(APPEND DEFAULT_CUDA_ARCH "60")
