--- a/CMakeLists.txt	2021-12-16 19:34:52.031697862 +0000
+++ b/CMakeLists.txt	2021-12-18 20:43:58.800154919 +0000
@@ -103,6 +103,13 @@
 # Optional packages
-find_package(UTempter)
+option(WITH_UTEMPTER "Enable support for utempter" OFF)
+if (WITH_UTEMPTER)
+  find_package(UTempter REQUIRED)
+endif(WITH_UTEMPTER)
+
 if(LINUX)
-  find_package(SELinux)
-endif()
+  option(WITH_SELINUX "Enable support for selinux" OFF)
+  if (WITH_SELINUX)
+  	find_package(SELinux REQUIRED)
+  endif(WITH_SELINUX)
+endif(LINUX)
 
@@ -216,3 +223,3 @@
 
-if(SELINUX_FOUND)
+if(WITH_SELINUX)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DWITH_SELINUX")
@@ -223,3 +230,3 @@
 
-if(UTEMPTER_FOUND)
+if(WITH_UTEMPTER)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DWITH_UTEMPTER")
