--- a/Makefile	2017-06-11 00:14:41.000000000 +0100
+++ b/Makefile	2017-09-12 00:13:49.506276962 +0100
@@ -1,3 +1 @@
-COPT		= -O3 -Wall -Werror
-BIN		= /usr/local/bin
 VERSION		= 0.5
@@ -13,2 +11,3 @@
 install:
+		mkdir -p ${BIN}
 		install dterm ${BIN}
@@ -17,7 +16 @@
 		rm -f dterm speeds.h *.o *.bak
-
-package:	$(SOURCEFILES)
-		mkdir -p dterm-$(VERSION)
-		cp $(SOURCEFILES) dterm-$(VERSION)
-		tar cfvz dterm-$(VERSION).tgz dterm-$(VERSION)
-		rm -rf dterm-$(VERSION)
