diff -urN a/configure b/configure
--- a/configure	2015-04-20 09:53:46.547442891 +0100
+++ b/configure	2015-04-20 09:55:10.470856563 +0100
@@ -18306,7 +18306,7 @@
 		# Solaris: GCC compiler
 		CPPFLAGS="$CPPFLAGS -pthreads -D_REENTRANT"
 		LDFLAGS="$LDFLAGS -pthreads"
-		PION_OPT_FLAGS="-O2 -ggdb -Wall -Wno-strict-aliasing -DNDEBUG"
+		PION_OPT_FLAGS="-O2 -Wall -Wno-strict-aliasing -DNDEBUG"
 		PION_DEBUG_FLAGS="-O0 -ggdb -Wall -Wno-strict-aliasing -fkeep-inline-functions"
 	fi
 	PION_EXTERNAL_LIBS="-lsocket -lnsl -lposix4"
@@ -18317,7 +18317,7 @@
 	CPPFLAGS="$CPPFLAGS -D_REENTRANT"
 	#LDFLAGS="$LDFLAGS"
 	#PION_OPT_FLAGS="-O3 -ggdb -fomit-frame-pointer -fstrict-aliasing -fno-tree-pre -falign-loops -Wall -Wno-strict-aliasing -DNDEBUG"
-	PION_OPT_FLAGS="-O3 -ggdb -fomit-frame-pointer -fstrict-aliasing -Wall -Wno-strict-aliasing -DNDEBUG"
+	PION_OPT_FLAGS="-O3 -fomit-frame-pointer -fstrict-aliasing -Wall -Wno-strict-aliasing -DNDEBUG"
 	PION_DEBUG_FLAGS="-O0 -ggdb -Wall -Wno-strict-aliasing -fkeep-inline-functions"
 
 	# Check for OSX binary architecture
@@ -18355,7 +18355,7 @@
 	# Cygwin GCC compiler (Windows)
 	CPPFLAGS="$CPPFLAGS -mthreads -D_REENTRANT -DPION_WIN32 -D__USE_W32_SOCKETS"
 	LDFLAGS="$LDFLAGS -mthreads -Wl,-E"
-	PION_OPT_FLAGS="-O2 -ggdb -Wall -Wno-strict-aliasing -DNDEBUG"
+	PION_OPT_FLAGS="-O2 -Wall -Wno-strict-aliasing -DNDEBUG"
 	PION_DEBUG_FLAGS="-O0 -ggdb -Wall -Wno-strict-aliasing -fkeep-inline-functions"
 	PION_EXTERNAL_LIBS="-lws2_32 -lmswsock"
 	;;
@@ -18363,7 +18363,7 @@
 	# FreeBSD: GCC compiler
 	CPPFLAGS="$CPPFLAGS -pthread -D_REENTRANT"
 	LDFLAGS="$LDFLAGS -pthread -Wl,-E"
-	PION_OPT_FLAGS="-O2 -ggdb -Wall -Wno-strict-aliasing -DNDEBUG"
+	PION_OPT_FLAGS="-O2 -Wall -Wno-strict-aliasing -DNDEBUG"
 	PION_DEBUG_FLAGS="-O0 -ggdb -Wall -Wno-strict-aliasing -fkeep-inline-functions"
 	;;
 *)
@@ -18371,7 +18371,7 @@
 	CPPFLAGS="$CPPFLAGS -pthread -D_REENTRANT"
 	LDFLAGS="$LDFLAGS -pthread -Wl,-E"
 	LIBS="-lrt -ldl"
-	PION_OPT_FLAGS="-O2 -ggdb -Wall -Wno-strict-aliasing -DNDEBUG"
+	PION_OPT_FLAGS="-O2 -Wall -Wno-strict-aliasing -DNDEBUG"
 	PION_DEBUG_FLAGS="-O0 -ggdb -Wall -Wno-strict-aliasing -fkeep-inline-functions"
 	;;
 esac
