--- a/python/mozterm/mozterm/terminal.py	2018-07-31 14:21:02.000000000 -0400
+++ b/python/mozterm/mozterm/terminal.py	2018-08-15 17:00:45.540515437 -0400
@@ -32,18 +32,9 @@
                 return ''
             return args[0]
 
     def __getattr__(self, attr):
         return self.NullCallableString()
 
 
 def Terminal(raises=False, disable_styling=False, **kwargs):
-    if disable_styling:
-        return NullTerminal(**kwargs)
-
-    try:
-        import blessings
-    except Exception:
-        if raises:
-            raise
-        return NullTerminal(**kwargs)
-    return blessings.Terminal(**kwargs)
+    return NullTerminal(**kwargs)
