--- cpuid-20160814/Makefile
+++ cpuid-20160814/Makefile
@@ -1,9 +1,3 @@
-CFLAGS+=-g
-CPPFLAGS?=
-LDFLAGS?=
-ifneq (,$(findstring arch=i386,$(CFLAGS)))
-CISA=-m32
-endif
 CFL=$(CPPFLAGS) $(CFLAGS) $(CISA) -Wall -Wshadow -Wcast-align -Wredundant-decls -Wbad-function-cast -Wcast-qual -Wwrite-strings -Waggregate-return -Wstrict-prototypes -Wmissing-prototypes -D_FILE_OFFSET_BITS=64 -DVERSION=$(VERSION)
 
 PACKAGE=cpuid
