diff -u1rN a/ipc/app/moz.build b/ipc/app/moz.build
--- a/ipc/app/moz.build	2017-04-11 03:13:12.000000000 +0100
+++ b/ipc/app/moz.build	2018-03-30 03:46:57.892450038 +0100
@@ -74,3 +74,3 @@
     if '-flto' in CONFIG['OS_CXXFLAGS'] and not CONFIG['CLANG_CXX']:
-        LDFLAGS += ['--param lto-partitions=1']
+        LDFLAGS += ['--lto-partition=one']
 
diff -u1rN a/ipc/app/pie/moz.build b/ipc/app/pie/moz.build
--- a/ipc/app/pie/moz.build	2017-04-11 03:13:12.000000000 +0100
+++ b/ipc/app/pie/moz.build	2018-03-30 03:46:57.893450050 +0100
@@ -27,3 +27,3 @@
     if '-flto' in CONFIG['OS_CXXFLAGS'] and not CONFIG['CLANG_CXX']:
-	LDFLAGS += ['--param lto-partitions=1']
+	LDFLAGS += ['--lto-partition=one']
 
