--- a/src/winetricks	2018-06-06 13:48:21.274833665 +0100
+++ a/src/winetricks	2018-06-06 13:57:02.733665022 +0100
@@ -4811,4 +4811,11 @@ winetricks_set_wineprefix()
         case "$WINE" in
-            *64) WINE64="${WINE}" ;;
-            *) WINE64="${WINE}64" ;;
+            *64)
+				WINE64="${WINE}"
+				;;
+            *)
+				WINE64="${WINE}64"
+				if [ ! -e "${WINE64}" ]; then
+					WINE64="${WINE/wine/wine64}"
+				fi
+				;;
         esac
