--- a/src/winetricks	2018-05-14 21:42:25.570682959 +0100
+++ b/src/winetricks	2018-05-14 21:45:07.193103656 +0100
@@ -3158,26 +3158,2 @@
 
-winetricks_detect_gui()
-{
-    if test -x "$(which zenity 2>/dev/null)"; then
-        WINETRICKS_GUI=zenity
-        WINETRICKS_GUI_VERSION="$(zenity --version)"
-        WINETRICKS_MENU_HEIGHT=500
-        WINETRICKS_MENU_WIDTH=1010
-    elif test -x "$(which kdialog 2>/dev/null)"; then
-        echo "Zenity not found!  Using kdialog as poor substitute."
-        WINETRICKS_GUI=kdialog
-        WINETRICKS_GUI_VERSION="$(kdialog --version)"
-    else
-        echo "No arguments given, so tried to start GUI, but zenity not found."
-        echo "Please install zenity if you want a graphical interface, or "
-        echo "run with --help for more options."
-        exit 1
-    fi
-
-    # Print zenity/dialog version info for debugging:
-    if [ ! "$WINETRICKS_SUPER_QUIET" ] ; then
-       echo "winetricks GUI enabled, using $WINETRICKS_GUI $WINETRICKS_GUI_VERSION"
-    fi
-}
-
 # Detect which sudo to use
@@ -5115,3 +5091,2 @@
     --force           Nicht prüfen ob Pakete bereits installiert wurden
-    --gui             GUI Diagnosen anzeigen, auch wenn von der Kommandozeile gestartet
     --isolate         Jedes Programm oder Spiel in eigener Bottle (WINEPREFIX) installieren
@@ -5152,3 +5127,2 @@
     --force           Don't check whether packages were already installed
-    --gui             Show gui diagnostics even when driven by commandline
     --isolate         Install each app or game in its own bottle (WINEPREFIX)
@@ -5191,3 +5165,2 @@
         --force) WINETRICKS_FORCE=1;;
-        --gui) winetricks_detect_gui;;
         -h|--help) winetricks_usage ; exit 0 ;;
@@ -5227,2 +5200,3 @@
     # Handle options before init, to avoid starting wine for --help or --version
+    [[ -z "${1}" ]] && winetricks_handle_option "-h"
     while winetricks_handle_option "$1"
@@ -19672,3 +19646,2 @@
             # No non-option arguments given, so read them from GUI, and loop until user quits
-            winetricks_detect_gui
             winetricks_detect_sudo
