--- a/toolkit/system/unixproxy/nsUnixSystemProxySettings.cpp	2022-07-01 23:48:29.380852674 +0100
+++ b/toolkit/system/unixproxy/nsUnixSystemProxySettings.cpp	2022-07-02 21:43:31.811215578 +0100
@@ -12,6 +12,7 @@
 #include "nsInterfaceHashtable.h"
 #include "nsHashtablesFwd.h"
 #include "nsHashKeys.h"
+#include "nsPrintfCString.h"
 #include "nsNetUtil.h"
 #include "nsISupportsPrimitives.h"
 #include "nsIGSettingsService.h"
