diff -urN a/mozilla/config/system-headers b/mozilla/config/system-headers
--- a/mozilla/config/system-headers	2015-12-09 01:35:02.806171625 +0000
+++ b/mozilla/config/system-headers	2015-12-09 02:21:10.827492222 +0000
@@ -402,6 +402,7 @@
 frame/req.h
 freetype/freetype.h
 freetype/ftcache.h
+freetype/ftfntfmt.h
 freetype/ftglyph.h
 freetype/ftsynth.h
 freetype/ftoutln.h
@@ -415,6 +416,7 @@
 freetype/ftxf86.h
 freetype.h
 ftcache.h
+ftfntfmt.h
 ftglyph.h
 ftsynth.h
 ftoutln.h
